transforms/transforms.go

```diff
@@ -215,3 +214,32 @@ func Unique(hash RowIdentifier) optimus.TransformFunc {
 		return false, nil
 	})
 }
+
+// GroupBy returns a TransformFunc that returns Rows of Rows grouped by id, which must be comparable.
+// Example:
```

Comment:
Maybe instead of giving an input example, saying something like:

> Each output row is one group of rows. The output row has two fields: id, which is the identifier for that group, and rows, which is the slice of Rows that share that identifier. For example, one row in a grouping by the "group" field might look like: {"id": "a", "rows": []optimus.Row{{"group": "a", "val": 2"}, {"group": "a", "val": 3}}}